src/citrine/resources/gemd_resource.py

```diff
@@ -106,18 +106,10 @@ def register_all(self, models: List[DataConcepts], *, dry_run=False) -> List[Dat
         for obj in models:
             by_type[obj.typ].append(obj)
         typ_groups = sorted(list(by_type.values()), key=lambda x: writable_sort_order(x[0]))
-        batch_size = 50
         for typ_group in typ_groups:
-            num_batches = len(typ_group) // batch_size
-            for batch_num in range(num_batches + 1):
-                batch = typ_group[batch_num * batch_size: (batch_num + 1) * batch_size]
-                if batch:  # final batch is empty when batch_size divides len(typ_group)
-                    registered = self._collection_for(batch[0]) \
-                        .register_all(batch, dry_run=dry_run)
-                    for prewrite, postwrite in zip(batch, registered):
-                        if isinstance(postwrite, BaseEntity):
-                            prewrite.uids = postwrite.uids
-                    resources.extend(registered)
+            registered = self._collection_for(typ_group[0]). \
+                register_all(typ_group, dry_run=dry_run)
```

Comment:
The line break doesn't seem necessary here.

```suggestion
            registered = self._collection_for(typ_group[0]).register_all(
                typ_group, dry_run=dry_run
            )
```